tests/loggers/test_wandb_logger.py

```diff
@@ -296,14 +309,18 @@ def test_wandb_artifacts(rank_zero_only: bool, tmp_path: pathlib.Path, dummy_sta
 
     # Wait for rank 0 queue the file upload
     dist.barrier()
-
     # Attempt to retrieve the artifact on all ranks
     downloaded_wandb_artifact_path = tmp_path / 'downloaded_wandb_artifact'
 
-    @retry(FileNotFoundError, num_attempts=6)  # 6 attempts is ~2^(6-1) seconds max wait
+    def mock_artifact(*arg, **kwargs):
+        return MockArtifact(dummy_wandb_artifact_path)
+
+    monkeypatch.setattr(Api, "artifact", mock_artifact)
+
     def _validate_wandb_artifact():
         wandb_logger.download_file(wandb_artifact_name, str(downloaded_wandb_artifact_path))
         with open(downloaded_wandb_artifact_path, 'r') as f:
             assert f.read() == 'hello!'
-
-    _validate_wandb_artifact()
+            
+    if dist.get_global_rank() == 0:
```

Comment:
the test docstring currently says that this test tests that an artifact uploaded from rank 0 is accessible on all ranks. doesn't this check make the test not do that?

Reply:
The artifact can't be accessed in offline mode, so I'm mocking the download portion by copying the file from the upload location to the download location. The file system is local to rank 0 though (since it was uploaded from rank 0) so only rank 0 will have access to the download. I also changed the docstring, but let me know if there's another way to test this on all ranks.

Reply:
yeah, id recommend syncing the artifact path between ranks, since they all have access to the same file system. they will just have different `tmp_path`s

Reply:
Done.